framework/test/unit/node/forger/forger.spec.ts

```diff
@@ -928,6 +928,38 @@ describe('forger', () => {
 				);
 			});
 
+			it('should log message and return if validator is not present for given time', async () => {
+				jest.spyOn(chainModuleStub, 'getValidator').mockResolvedValue(undefined);
+
+				const today = new Date();
+				const future = new Date(today.getTime() + 60 * 60 * 100);
+				const futureTimestamp = future.getTime();
+				const futureSlotTime = Math.floor(futureTimestamp / 1000);
+
+				const dateNowMockFn = jest
+					.spyOn(Date.prototype, 'getTime')
+					.mockReturnValue(futureTimestamp);
+
+				chainModuleStub.slots.getSlotTime.mockReturnValue(futureSlotTime);
+
+				const changedLastBlockSlot = currentSlot - 2;
+				when(getSlotNumberStub)
+					.calledWith(lastBlock.header.timestamp)
+					.mockReturnValue(changedLastBlockSlot);
+
+				chainModuleStub.slots.getSlotTime.mockReturnValue(futureSlotTime);
+
+				const data = await forgeModule.forge();
+
+				expect(data).toBeUndefined();
+				expect(loggerStub.trace).toHaveBeenCalledTimes(1);
```

Comment:
I think it's better not to check the log. It's better to check if the block is not created and no error is thrown?